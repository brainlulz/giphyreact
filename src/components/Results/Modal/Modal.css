.Modal {
  background-color: rgba(251, 248, 254, 0.45);
  position: absolute;
  z-index: 1;
  width: 224px;
  height: 240px;
  border: 0.5rem solid transparent;
  display: grid;
  justify-content: center;
  align-items: center;
  opacity: 0;
  transition: opacity 0.5s ease
}

.FavBtn {
  background-color: transparent;
  border: none;
  background-image: url("https://image.noelshack.com/fichiers/2018/09/2/1519761895-coeur.png");
  display: inline-block;
  width: 3rem;
  height: 3rem;
  background-size: 100%;
  z-index: 2;
  transition: transform 0.1s ease-in-out;
}

.FavBtn:hover {
  transform: scale(2, 2);
}

.FavBtn a {
  color: #930077;
  font-size: 1.5rem;
}