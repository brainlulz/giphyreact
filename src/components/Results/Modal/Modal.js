import React from "react";

import "./Modal.css";

const Modal = ({ gif, onToggleFavorite, hrefLink }) => {
  return (
    <div className="Modal">
      <button className="FavBtn" onClick={() => onToggleFavorite(gif)} id={gif.id}/>
      <a href={hrefLink} target="_blank" rel="noopener noreferrer">
        Link
      </a>
    </div>
  );
};
export default Modal;
