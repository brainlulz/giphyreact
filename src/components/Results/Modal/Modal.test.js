import React from 'react';
import Modal from './index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Modal gif='' onToggleFavorite='' hrefLink={true}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});