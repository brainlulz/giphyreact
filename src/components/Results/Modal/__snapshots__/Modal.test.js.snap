// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Modal"
>
  <button
    className="FavBtn"
    id={undefined}
    onClick={[Function]}
  />
  <a
    href={true}
    rel="noopener noreferrer"
    target="_blank"
  >
    Link
  </a>
</div>
`;
